### PR TITLE
Eagerly push metrics when pushInterval is 0

### PR DESCRIPTION
**Pull request by brettimus**
Implements #98 

***

Tested with a Supabase Edge Function locally and it works.

(To test with an edge function, I published the changes to the core lib as a package called `brettermetrics`on npm, then imported in Deno via `https://esm.sh/brettermetrics@0.6.0`)

***

Note that this is only a hotfix for _experimental_ support in FaaS environments. We can do a follow-up PR to hash out things like configuration, error logging, etc.

### CODE REVIEW

File: autometrics-ts.code-workspace [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+		{
+			"path": "."
+		}
+	],
+	"settings": {
+		"editor.defaultFormatter": "rome.rome",
+		"editor.formatOnSave": true,
+		"typescript.tsdk": "node_modules/typescript/lib"
+	}
+}
```
Comment (arendjr) [inline autometrics-ts.code-workspace:12]: I think this shouldn't be part of the repository.

File: packages/lib/src/instrumentation.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ import {
 import { buildInfo, BuildInfo, recordBuildInfo } from "./buildInfo";
 import { HISTOGRAM_NAME } from "./constants";
 
+let IS_EAGERLY_PUSHED = false;
+let _pushMetrics = () => {};
```
Comment (arendjr) [inline packages/lib/src/instrumentation.ts:18]: What's up with the underscore?

File: packages/lib/src/instrumentation.ts [hunk 3]
```diff
@@ -14,6 +14,8 @@ import {
 import { buildInfo, BuildInfo, recordBuildInfo } from "./buildInfo";
 import { HISTOGRAM_NAME } from "./constants";
 
+let IS_EAGERLY_PUSHED = false;
```
Comment (arendjr) [inline packages/lib/src/instrumentation.ts:17]: Why the ALL_CAPS? That's usually used for constants, which this is not.

File: packages/lib/src/instrumentation.ts [hunk 4]
```diff
@@ -80,19 +90,74 @@ export function init(options: initOptions) {
...
+      // we flush the metrics at the end of the submission to ensure the data is not repeated
+      await exporter.forceFlush();
+    } catch (error) {
+      console.error(
+        `Error flushing metrics after push: ${
+          error?.message ?? "<no error message found>"
+        }`,
+      );
+    }
+  } catch (collectError) {
```
Comment (arendjr) [inline packages/lib/src/instrumentation.ts:154]: Why the nested try-catches? If you expect this catch to only apply to the collecting part, why not place it *before* the other two try-catch blocks?

**Review by arendjr — APPROVED** [review]
Functionality looks good, just some style comments.

File: examples/faas-experimental/fetch-polyfill.js
Comment (keturiosakys) [inline examples/faas-experimental/fetch-polyfill.js:1]: comment-more-of-a-question: don't all modern FaaS environments have `fetch` by default? Thinking especially of Deno in this case.

**Review by keturiosakys — APPROVED** [review]
LGTM!

File: autometrics-ts.code-workspace (hunk 1 above)
Comment (brettimus) [reply autometrics-ts.code-workspace:12]: ah yeah, i'll remove

File: packages/lib/src/instrumentation.ts (hunk 3 above)
Comment (brettimus) [reply packages/lib/src/instrumentation.ts]: my personal convention for local global variables — changing it!

File: packages/lib/src/instrumentation.ts (hunk 2 above)
Comment (brettimus) [reply packages/lib/src/instrumentation.ts]: benno knows — i love underscores

another personal convention for something that can be reassigned across multiple functions, will remove

File: examples/faas-experimental/fetch-polyfill.js
Comment (brettimus) [reply examples/faas-experimental/fetch-polyfill.js:1]: ah yeah - i think this will be a non issue if the environment runs on node 18 or later

**Pull request merged**